src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/AuditingSettings/SetAzureSqlServerAuditing.cs

```diff
@@ -54,6 +54,13 @@ public class SetAzureSqlServerAuditing : SqlServerAuditingSettingsCmdletBase
         [ValidateNotNullOrEmpty]
         public string StorageAccountName { get; set; }
 
+        /// <summary>
+        /// Gets or sets storage account subscription id.
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = AuditingHelpMessages.AuditStorageAccountSubscriptionIdHelpMessage)]
```

Comment:
As per the discussion in the [design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/99#issuecomment-397068036)

The behavior as the changes stand currently is that the user can specify `StorageAccountSubscriptionId` without specifying `StorageAccountName`.

Given that `StorageAccountName` is required when StorageAccountSubscriptionId` is specified, I would recomment adding a parameter set where `StorageAccountName` is required and `StorageAccountSubscriptionId` is optional.